devicetypes/axis/axis-gear-v2.src/axis-gear-v2.groovy

```diff
@@ -0,0 +1,196 @@
+                tileAttribute ("device.level", key: "VALUE_CONTROL") {
+              		attributeState("VALUE_UP", action: "ShadesUp")
+        			attributeState("VALUE_DOWN", action: "ShadesDown")
+             }
+   		}
+        //Added a "doubled" state to toggle states between positions
+        standardTile("main", "device.windowShade"){
+        	state("open", label:'Open', action:"close", icon:"http://i.imgur.com/St7oRQl.png", backgroundColor:"#ffcc33", nextState: "closed")
+            state("partial", label:'Partial',  icon:"http://i.imgur.com/y0ZpmZp.png", backgroundColor:"#ffcc33")
+            state("closed", label:'Closed',action:"open", icon:"http://i.imgur.com/SAiEADI.png", backgroundColor:"#bbbbdd", nextState: "open")
+        }
+	 	controlTile("mediumSlider", "device.level", "slider",decoration:"flat",height:1, width: 2, inactiveLabel: true) {
+            state("level", action:"setLevel")
+        }
+        
+        valueTile("battery", "device.battery", inactiveLabel:false, decoration:"flat", width:2, height:1) {
+			state "battery", label:'${currentValue}% battery', unit:""
+		}
+        
+		standardTile("refresh", "device.refresh", inactiveLabel:false, decoration:"flat", width:2, height:1) {
+			state "default", label:'', action:"refresh.refresh", icon:"st.secondary.refresh"
+		}
+        
+        main(["main"])
+        details(["shade", "mediumSlider","battery", "refresh"])
+        //details('shade', 'levelSliderControl')
+        //details('levelSliderControl', 'otherTile', 'anotherTile') //adjustment and order of tiles
+	}
+
+}
+//Declare Clusters
+private getCLUSTER_POWER() {0x0001}
+private getCLUSTER_LEVEL() {0x0008}
+private getLEVEL_ATTR_LEVEL() {0x0000}
+private getPOWER_ATTR_BATTERY() {0x0021}
+
+//Custom command to increment blind position by 25 %
+def ShadesUp(){
+	def shadeValue = device.latestValue("level") as Integer ?: 0 
+    
+    if (shadeValue < 100){
+      	shadeValue = Math.min(25 * (Math.round(shadeValue / 25) + 1), 100) as Integer
+    }else { 
+    	shadeValue = 100
+	}
+    //sendEvent(name:"level", value:shadeValue, displayed:true)
+    setLevel(shadeValue)
+}
+
+//Custom command to decrement blind position by 25 %
+def ShadesDown(){
+	def shadeValue = device.latestValue("level") as Integer ?: 0 
+    
+    if (shadeValue > 0){
+      	shadeValue = Math.max(25 * (Math.round(shadeValue / 25) - 1), 0) as Integer
+    }else { 
+    	shadeValue = 0
+	}
+    //sendEvent(name:"level", value:shadeValue, displayed:true)
+    setLevel(shadeValue)
+	   
+}
+
+//Send Command through setLevel()
+def on() {
+	//sendEvent(name:"level", value:0, displayed:true)
+    setLevel(100)  
+}
+
+//Send Command through setLevel()
+def off() {
+	setLevel(0)
+}
+//Command to set the blind position (%) and log the event
+def setLevel(value) {
+	sendEvent(name:"level", value: value, displayed:true)
+    def L = Math.round(value);
+    def i = Integer.valueOf(L.intValue());
+    setWindowShade(i)
+	zigbee.setLevel(i)
```

Comment:
you should be able to call `zigbee.setLevel(value)` like we do in our other DTHs without the need for rounding or casting and calling `setWindowShade(value as int)` should do all of the casting you would need, I think.

Reply:
If you've configured reporting correctly, shouldn't the device report its level after it is set, making the call to `setWindowShade()` redundant?